src/utils/transport.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Transport interface should be used for this project only.
+ *
+ * `get` and `post` methods should return a Promise:
+ * - fullfilled with the response Json body as JavaScript object when the response status code is 200
+ * - rejected with an Error otherwise
+ */
+export interface Transport {
+  get<ResponseType>(url: string, headers: { [headerName: string]: string }): Promise<ResponseType>;
```

Comment:
I tried with a Map and it works too. I use objects because it has a simpler syntax. On documentation examples for fetch (using headers headers) they use objects (or Header object) https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch